custom_metrics/wpt_bodies.js

```diff
@@ -0,0 +1,1366 @@
+                                current_noopener = true;
+                                break;
+                              case "noreferrer":
+                                target.rel_attributes.noreferrer++;
+                                current_noreferrer = true;
+                                break;
+                          }
+                      });
+                  });
+              }
+
+              if (node.target) {
+                let targetAttribute = node.target.trim();
+
+
+                if (targetAttribute == "_blank") {
+                  target.target_blank.total++;
+
+                  if (current_noopener && current_noreferrer) {
+                    target.target_blank.noopener_noreferrer++;
+                  } else if (current_noopener) {
+                    target.target_blank.noopener++;
+                  } else if (current_noreferrer) {
+                    target.target_blank.noreferrer++;
+                  } else {
+                    target.target_blank.neither++;
+                  }
+                }
+
+                if (target.targets[targetAttribute]) 
+                  target.targets[targetAttribute]++;
+                else 
+                  target.targets[targetAttribute] = 1;
+              }
+
+              // see if it is an image link
+              // no visible text
+              let noText = node.innerText.trim().length === 0;
+              let hasImage = node.querySelector('img') !== null;
+
+              if (noText) {
+                  if (hasImage) {
+                    target.image_links++;
+                  }
+                  else {
+                      // invisible link? 
+                      target.invisible_links++;
+                  }
+              }
+              else {
+                target.text_links++;
+              }
+
+              if (crawlable) { // unless nofollow ???
+                if (follow) {
+                  target.crawlable.follow++;
+                } else
+                {
+                  target.crawlable.nofollow++;
+                }
+              }
+
+            });
+          }
+          return target;
+        };
+
+        let result = {};
+
+        result.rendered = getAnchorData(document)
+
+        let rawHtmlDocument = getRawHtmlDocument();
+
+        if (rawHtmlDocument) {
+          result.raw = getAnchorData(rawHtmlDocument);
+        }
+
+        return result;
+      }
+      catch(e) {
+        return logError("anchors", e);
+      }
+    })(),
+
+
+    // Extract the real title tag contents  
+    // Used by: SEO
+    // Backup: old version in almanac.js 
+    'title': (() => {
+      try {    
+        let result = {};
+
+        function getTitles(d) {
+          let target = {};
+          target.total = Array.from(d.querySelectorAll('head title')).map(e => {
+            let text = e.innerText.trim();
+            let characters = text.length;
+            let words = text.match(/\S+/g)?.length;
```

Comment:
Curious, how does this behave for non-English languages or non-Latin alphabets?

Similarly for `characters`, could it be double-counting 2-byte characters?